### PR TITLE
5 : WIP Evaluation using UserVideo

### PR DESCRIPTION
**Pull request by ShaneKoNaung**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -128,4 +128,315 @@ python ss.py -w -is -v ~/Desktop/ff2-screenrecord-first4min.mov
...
+Filename: FF2_EN_1, Predicted: 0, Ground Truth: o
+Filename: FF2_EN_1, Predicted: o, Ground Truth: a
+Filename: FF2_EN_1, Predicted: c, Ground Truth: C
+Filename: FF2_EN_1, Predicted: . 1 , Ground Truth:
+Filename: FF2_EN_5, Predicted: o, Ground Truth: a
+Filename: FF2_EN_5, Predicted: , Ground Truth: i
+Filename: FF2_EN_5, Predicted: o, Ground Truth: a
+Filename: FF2_EN_5, Predicted:  , Ground Truth:
+Filename: FF2_EN_6, Predicted: d I, Ground Truth: all
+Filename: FF2_EN_6, Predicted: 0, Ground Truth: o
```
Comment (mattkanwisher) [inline README.md]: Do we really need all this stuff in the README.md?

File: eval.sh [hunk 2]
```diff
@@ -0,0 +1,15 @@
+#!/bin/bash
+
+
+
+# echo =========== Generation Predictions for English images using UserVideo ==========
+# python generate_predictions.py
+
```
Comment (mattkanwisher) [inline eval.sh]: can we move this code to a submodule?

File: user_video.py [hunk 3]
```diff
@@ -8,16 +8,37 @@
...
+                background_task_args={"translate": translate, 'enable_cache': enable_cache},
+                show_fps=True,
+                crop_y_coordinate=None,
+                frameProcessor=self.frameProcessor,
+                textDetector=textDetector,
+                debug_bbox=debug_bbox
+            )
+
+        if eval is True:
+            self.frameProcessor_jp = FrameProcessor(language='jp', disable_dialog=disable_dialog)
```
Comment (mattkanwisher) [inline user_video.py]: the evaluation code shouldn't leak into the production code. Can you just create 2 instances of user_video one with EN and one with JP?

File: user_video.py [hunk 4]
```diff
@@ -55,4 +76,38 @@ def get_immediate_frame(self):
...
+            return {
+                "image": str(image_path),
+                "text_detected": False,
+                "recognized_text": '',
+                "annotations": [],
+                "translation": ''
+            }
+        
+        if lang == 'EN':
+            closest_match, previous_image, highlighted_image, annotations, translation = self.frameProcessor_en.run_image(img, translate=self.video_stream.background_task_args.get("translate", None), enable_cache=self.video_stream.background_task_args.get("enable_cache", False))
```
Comment (mattkanwisher) [inline user_video.py]: can we keep a single instance to a single language and we don't need to do this frameprocess_en / frameprocessor_jp

File: README.md (hunk 1 above)
Comment (ShaneKoNaung) [reply README.md]: I've removed these.

File: eval.sh (hunk 2 above)
Comment (ShaneKoNaung) [reply eval.sh]: All the eval codes are refactored and moved into ocr_eval folder.

File: user_video.py (hunk 3 above)
Comment (ShaneKoNaung) [reply user_video.py]: I've reverted the user_video.py code to previous version. removed all the eval related code.

File: user_video.py (hunk 4 above)
Comment (ShaneKoNaung) [reply user_video.py]: I've changed the code for this.

**Pull request merged**